### ISSUE TITLE
Where are the command line options printed?

### ISSUE THREAD

**Issue by blattms** [OPM/opm-simulators#4410]
I am probably making a fool out of myself, but anyway:

What file did we print all the parameters used to? Somehow I (and grep) can't find them anymore.

**Comment by GitPaean** [comment]
I use the following, not sure it contains **all** the parameters. 

```
./flow_blackoil --help
Usage: ./flow_blackoil [OPTIONS] [ECL_DECK_FILENAME]
Flow, an advanced reservoir simulator for ECL-decks provided by the Open Porous Media project.

Recognized options:
    -h,--help                                     Print this help message and exit
    --help-all                                    Print all parameters, including obsolete, hidden and deprecated ones.
    --accelerator-mode=STRING                     Choose a linear solver, usage: '--accelerator-mode=[none|cusparse|opencl|amgcl|rocalution]'. Default: "none"
    --allow-distributed-wells=BOOLEAN             Allow the perforations of a well to be distributed to interior of multiple processes. Default: false
    --alternative-well-rate-init=BOOLEAN          Use alternative well rate initialization procedure. Default: true
    --bda-device-id=INTEGER                       Choose device ID for cusparseSolver or openclSolver, use 'nvidia-smi' or 'clinfo' to determine valid IDs. Default: 0
    --continue-on-convergence-error=BOOLEAN       Continue with a non-converged solution instead of giving up if we encounter a time step size smaller than the minimum time step size. Default: false
    --cpr-reuse-interval=INTEGER                  Reuse preconditioner interval. Used when CprReuseSetup is set to 4, then the preconditioner will be fully recreated instead of reused every N linear solve, where
                                                    N is this parameter. Default: 30
    --cpr-reuse-setup=INTEGER                     Reuse preconditioner setup. Valid options are 0: recreate the preconditioner for every linear solve, 1: recreate once every timestep, 2: recreate if last linear
                                                    solve took more than 10 iterations, 3: never recreate, 4: recreated every CprReuseInterval. Default: 4
    --dbhp-max-rel=SCALAR                         Maximum relative change of the bottom-hole pressure in a single iteration. Default: 1
    --dp-max-rel=SCALAR                           Maximum relative change of pressure in a single iteration. Default: 0.3
    --ds-max=SCALAR                               Maximum absolute change of any saturation in a single iteration. Default: 0.2
    --dwell-fraction-max=SCALAR                   Maximum absolute change of a well's volume fraction in a single iteration. Default: 0.2
    --ecl-deck-file-name=STRING                   The name of the file which contains the ECL deck to be simulated. Default: ""
    --ecl-enable-drift-compensation=BOOLEAN       Enable partial compensation of systematic mass losses via the source term of the next time step. Default: true
    --ecl-output-double-precision=BOOLEAN         Tell the output writer to use double precision. Useful for 'perfect' restarts. Default: false
    --ecl-output-interval=INTEGER                 The number of report steps that ought to be skipped between two writes of ECL results. Default: -1
    --ecl-strict-parsing=BOOLEAN                  Use strict mode for parsing - all errors are collected before the applicaton exists. Default: false
    --edge-weights-method=INTEGER                 Choose edge-weighing strategy: 0=uniform, 1=trans, 2=log(trans). Default: 1
    --enable-adaptive-time-stepping=BOOLEAN       Use adaptive time stepping between report steps. Default: true
    --enable-async-ecl-output=BOOLEAN             Write the ECL-formated results in a non-blocking way (i.e., using a separate thread). Default: true
    --enable-async-vtk-output=BOOLEAN             Dispatch a separate thread to write the VTK output. Default: true
    --enable-dry-run=STRING                       Specify if the simulation ought to be actually run, or just pretended to be. Default: "auto"
    --enable-ecl-output=BOOLEAN                   Write binary output which is compatible with the commercial Eclipse simulator. Default: true
    --enable-esmry=BOOLEAN                        Write ESMRY file for fast loading of summary data. Default: false
    --enable-logging-fallout-warning=BOOLEAN      Developer option to see whether logging was on non-root processors. In that case it will be appended to the *.DBG or *.PRT files. Default: false
    --enable-opm-rst-file=BOOLEAN                 Include OPM-specific keywords in the ECL restart file to enable restart of OPM simulators from these files. Default: false
    --enable-storage-cache=BOOLEAN                Store previous storage terms and avoid re-calculating them. Default: true
    --enable-terminal-output=BOOLEAN              Print high-level information about the simulation's progress to the terminal. Default: true
    --enable-tuning=BOOLEAN                       Honor some aspects of the TUNING keyword. Default: false
    --enable-vtk-output=BOOLEAN                   Global switch for turning on writing VTK files. Default: false
    --enable-well-operability-check=BOOLEAN       Enable the well operability checking. Default: true
    --enable-well-operability-check-iter=BOOLEAN  Enable the well operability checking during iterations. Default: false
    --enable-write-all-solutions=BOOLEAN          Write all solutions to disk instead of only the ones for the report steps. Default: false
    --force-disable-fluid-in-place-output=BOOLEAN  Do not print fluid-in-place values after each report step even if requested by the deck. Default: false
    --force-disable-resv-fluid-in-place-output=BOOLEAN  Do not print reservoir volumes values after each report step even if requested by the deck. Default: false
    --full-time-step-initially=BOOLEAN            Always attempt to finish a report step using a single substep. Default: false
    --ignore-keywords=STRING                      List of Eclipse keywords which should be ignored. As a ':' separated string. Default: ""
    --ilu-fillin-level=INTEGER                    The fill-in level of the linear solver's ILU preconditioner. Default: 0
    --ilu-redblack=BOOLEAN                        Use red-black partitioning for the ILU preconditioner. Default: false
    --ilu-relaxation=SCALAR                       The relaxation factor of the linear solver's ILU preconditioner. Default: 0.9
    --ilu-reorder-spheres=BOOLEAN                 Whether to reorder the entries of the matrix in the red-black ILU preconditioner in spheres starting at an edge. If false the original ordering is preserved in
                                                    each color. Otherwise why try to ensure D4 ordering (in a 2D structured grid, the diagonal elements are consecutive). Default: false
    --initial-time-step-in-days=SCALAR            The size of the initial time step in days. Default: 1
    --linear-solver=STRING                        Configuration of solver. Valid options are: ilu0 (default), cprw, cpr (an alias for cprw), cpr_quasiimpes, cpr_trueimpes or amg. Alternatively, you can request a
                                                    configuration to be read from a JSON file by giving the filename here, ending with '.json.'. Default: "ilu0"
    --linear-solver-ignore-convergence-failure=BOOLEAN  Continue with the simulation like nothing happened after the linear solver did not converge. Default: false
    --linear-solver-max-iter=INTEGER              The maximum number of iterations of the linear solver. Default: 200
    --linear-solver-reduction=SCALAR              The minimum reduction of the residual which the linear solver must achieve. Default: 0.01
    --linear-solver-restart=INTEGER               The number of iterations after which GMRES is restarted. Default: 40
    --linear-solver-verbosity=INTEGER             The verbosity level of the linear solver (0: off, 2: all). Default: 0
    --matrix-add-well-contributions=BOOLEAN       Explicitly specify the influences of wells between cells in the Jacobian and preconditioner matrices. Default: false
    --max-inner-iter-ms-wells=INTEGER             Maximum number of inner iterations for multi-segment wells. Default: 100
    --max-inner-iter-wells=INTEGER                Maximum number of inner iterations for standard wells. Default: 50
    --max-newton-iterations-with-inner-well-iterations=INTEGER  Maximum newton iterations with inner well iterations. Default: 8
    --max-pressure-change-ms-wells=SCALAR         Maximum relative pressure change for a single iteration of the multi-segment well model. Default: 1e+06
    --max-residual-allowed=SCALAR                 Absolute maximum tolerated for residuals without cutting the time step size. Default: 1e+07
    --max-single-precision-days=SCALAR            Maximum time step size where single precision floating point arithmetic can be used solving for the linear systems of equations. Default: 20
    --max-temperature-change=SCALAR               Maximum absolute change of temperature in a single iteration. Default: 5
    --max-welleq-iter=INTEGER                     Maximum number of iterations to determine solution the  well equations. Default: 30
    --maximum-number-of-well-switches=INTEGER     Maximum number of times a well can switch to the same control. Default: 3
    --milu-variant=STRING                         Specify which variant of the modified-ILU preconditioner ought to be used. Possible variants are: ILU (default, plain ILU), MILU_1 (lump diagonal with dropped row
                                                    entries), MILU_2 (lump diagonal with the sum of the absolute values of the dropped row  entries), MILU_3 (if diagonal is positive add sum of dropped row
                                                    entrires. Otherwise subtract them), MILU_4 (if diagonal is positive add sum of dropped row entrires. Otherwise do nothing. Default: "ILU"
    --min-strict-cnv-iter=INTEGER                 Minimum number of Newton iterations before relaxed tolerances can be used for the CNV convergence criterion. Default: 0
    --min-time-step-based-on-newton-iterations=SCALAR  The minimum time step size (in days for field and metric unit and hours for lab unit) can be reduced to based on newton iteration counts. Default: 0
    --min-time-step-before-shutting-problematic-wells-in-days=SCALAR  The minimum time step size in days for which problematic wells are not shut. Default: 0.01
    --newton-max-iterations=INTEGER               The maximum number of Newton iterations per time step. Default: 20
    --newton-max-relax=SCALAR                     The maximum relaxation factor of a Newton iteration. Default: 0.5
    --newton-min-iterations=INTEGER               The minimum number of Newton iterations per time step. Default: 1
    --newton-relaxation-type=STRING               The type of relaxation used by Newton method. Default: "dampen"
    --num-jacobi-blocks=INTEGER                   Number of blocks to be created for the Block-Jacobi preconditioner. Default: 0
    --opencl-ilu-parallel=BOOLEAN                 Parallelize ILU decomposition and application on GPU. Default: true
    --opencl-platform-id=INTEGER                  Choose platform ID for openclSolver, use 'clinfo' to determine valid platform IDs. Default: 0
    --output-dir=STRING                           The directory to which result files are written. Default: ""
    --output-extra-convergence-info=STRING        Provide additional convergence output files for diagnostic purposes. "none" gives no extra output and overrides all other options, "steps" generates an INFOSTEP
                                                    file, "iterations" generates an INFOITER file. Combine options with commas, e.g., "steps,iterations" for multiple outputs. Default: "none"
    --output-interval=INTEGER                     Specify the number of report steps between two consecutive writes of restart data. Default: 1
    --output-mode=STRING                          Specify which messages are going to be printed. Valid values are: none, log, all (default). Default: "all"
    --owner-cells-first=BOOLEAN                   Order cells owned by rank before ghost/overlap cells. Default: true
    --parameter-file=STRING                       An .ini file which contains a set of run-time parameters. Default: ""
    --pri-var-oscilation-threshold=SCALAR         The threshold value for the primary variable switching conditions after its meaning has switched to hinder oscilations. Default: 1e-05
    --print-parameters=INTEGER                    Print the values of the run-time parameters at the start of the simulation. Default: 2
    --print-properties=INTEGER                    Print the values of the compile time properties at the start of the simulation. Default: 2
    --project-saturations=BOOLEAN                 Option for doing saturation projection. Default: false
    --regularization-factor-wells=SCALAR          Regularization factor for wells. Default: 100
    --relaxed-max-pv-fraction=SCALAR              The fraction of the pore volume of the reservoir where the volumetric error (CNV) may be voilated during strict Newton iterations. Default: 0.03
    --relaxed-pressure-tol-msw=SCALAR             Relaxed tolerance for the MSW pressure solution. Default: 10000
    --relaxed-well-flow-tol=SCALAR                Relaxed tolerance for the well flow residual. Default: 0.001
    --scale-linear-system=BOOLEAN                 Scale linear system according to equation scale and primary variable types. Default: false
    --sched-restart=BOOLEAN                       When restarting: should we try to initialize wells and groups from historical SCHEDULE section. Default: false
    --serial-partitioning=BOOLEAN                 Perform partitioning for parallel runs on a single process. Default: false
    --shut-unsolvable-wells=BOOLEAN               Shut unsolvable wells. Default: true
    --solve-welleq-initially=BOOLEAN              Fully solve the well equations before each iteration of the reservoir model. Default: true
    --solver-continue-on-convergence-failure=BOOLEAN  Continue instead of stop when minimum solver time step is reached. Default: false
    --solver-growth-factor=SCALAR                 The factor time steps are elongated after a successful substep. Default: 2
    --solver-max-growth=SCALAR                    The maximum factor time steps are elongated after a report step. Default: 3
    --solver-max-restarts=INTEGER                 The maximum number of breakdowns before a substep is given up and the simulator is terminated. Default: 10
    --solver-max-time-step-in-days=SCALAR         The maximum size of a time step in days. Default: 365
    --solver-min-time-step=SCALAR                 The minimum size of a time step in days for field and metric and hours for lab. If a step cannot converge without getting cut below this step size the simulator
                                                    will stop. Default: 1e-12
    --solver-restart-factor=SCALAR                The factor time steps are elongated after restarts. Default: 0.33
    --solver-verbosity=INTEGER                    Specify the "chattiness" of the non-linear solver itself. Default: 1
    --strict-inner-iter-wells=INTEGER             Number of inner well iterations with strict tolerance. Default: 40
    --strict-outer-iter-wells=INTEGER             Number of newton iterations for which wells are checked with strict tolerance. Default: 6
    --temperature-max=SCALAR                      Maximum absolute temperature. Default: 1e+09
    --temperature-min=SCALAR                      Minimum absolute temperature. Default: 0
    --threads-per-process=INTEGER                 The maximum number of threads to be instantiated per process ('-1' means 'automatic'). Default: -1
    --time-step-after-event-in-days=SCALAR        Time step size of the first time step after an event occurs during the simulation in days. Default: -1
    --time-step-control=STRING                    The algorithm used to determine time-step sizes. valid options are: 'pid' (default), 'pid+iteration', 'pid+newtoniteration', 'iterationcount',
                                                    'newtoniterationcount' and 'hardcoded'. Default: "pid+newtoniteration"
    --time-step-control-decay-damping-factor=SCALAR  The decay rate of the time step decrease when the target iterations is exceeded. Default: 1
    --time-step-control-decay-rate=SCALAR         The decay rate of the time step size of the number of target iterations is exceeded. Default: 0.75
    --time-step-control-file-name=STRING          The name of the file which contains the hardcoded time steps sizes. Default: "timesteps"
    --time-step-control-growth-damping-factor=SCALAR  The growth rate of the time step increase when the target iterations is undercut. Default: 3.2
    --time-step-control-growth-rate=SCALAR        The growth rate of the time step size of the number of target iterations is undercut. Default: 1.25
    --time-step-control-target-iterations=INTEGER  The number of linear iterations which the time step control scheme should aim for (if applicable). Default: 30
    --time-step-control-target-newton-iterations=INTEGER  The number of Newton iterations which the time step control scheme should aim for (if applicable). Default: 8
    --time-step-control-tolerance=SCALAR          The tolerance used by the time step size control algorithm. Default: 0.1
    --time-step-verbosity=INTEGER                 Specify the "chattiness" during the time integration. Default: 1
    --tolerance-cnv=SCALAR                        Local convergence tolerance (Maximum of local saturation errors). Default: 0.01
    --tolerance-cnv-relaxed=SCALAR                Relaxed local convergence tolerance that applies for iterations after the iterations with the strict tolerance. Default: 1
    --tolerance-mb=SCALAR                         Tolerated mass balance error relative to total mass present. Default: 1e-06
    --tolerance-pressure-ms-wells=SCALAR          Tolerance for the pressure equations for multi-segment wells. Default: 1000
    --tolerance-well-control=SCALAR               Tolerance for the well control equations. Default: 1e-07
    --tolerance-wells=SCALAR                      Well convergence tolerance. Default: 0.0001
    --update-equations-scaling=BOOLEAN            Update scaling factors for mass balance equations during the run. Default: false
    --use-gmres=BOOLEAN                           Use GMRES as the linear solver. Default: false
    --use-multisegment-well=BOOLEAN               Use the well model for multi-segment wells instead of the one for single-segment wells. Default: true
    --use-update-stabilization=BOOLEAN            Try to detect and correct oscillations or stagnation during the Newton method. Default: true
    --zoltan-imbalance-tol=SCALAR                 Tolerable imbalance of the loadbalancing provided by Zoltan (default: 1.1). Default: 1.1
```

**Comment by akva2** [comment]
i don't think we dump by default an longer, but there's a --print-parameters

**Comment by alfbr** [comment]
The parameters were dropped to the top of the PRT file, just have a look at an old run.

**Comment by GitPaean** [comment]
>  The parameters were dropped to the top of the PRT file, just have a look at an old run.

They are printed in different format, like 
`--enable-tuning=true` is printed as ` EnableTuning="true"`

not sure this is why Markus has grep difficulties?

**Comment by alfbr** [comment]
> > The parameters were dropped to the top of the PRT file, just have a look at an old run.
> 
> They are printed in different format, like `--enable-tuning=true` is printed as ` EnableTuning="true"`
> 
> not sure this is why Markus has grep difficulties?

Nope, I think the issue is that they are not printed at all to the PRT file now, it is a bug.

**Comment by GitPaean** [comment]
> Nope, I think the issue is that they are not printed at all to the PRT file now, it is a bug.

Yes, you are right. I am now seeing this with the new runs. Not sure when it started this way.

**Comment by alfbr** [comment]
It was between the 18th and 19th of January.

**Comment by bska** [comment]
>>> Nope, I think the issue is that they are not printed at all to the PRT file now, it is a bug.
>>
>> Yes, you are right. I am now seeing this with the new runs. Not sure when it started this way.
>
> It was between the 18th and 19th of January.

I think I've tracked down the root cause.  PR #4396 (commit 82d2146c9) moved the deck-reading code to a new `.cpp` file and appears to have omitted the call to [`FlowMainEbos<>::printPRTHeader()`](https://github.com/OPM/opm-simulators/blob/87a357f77e67a767cc4a3f0290d825e3176c097c/opm/simulators/flow/FlowMainEbos.hpp#L347) in the process.  I did not notice this whilst reviewing the PR, so sorry about that.

@akva2: How much work would it be to reinstate the `printPRTHeader()` call in an appropriate location?
